src/components/Editor/TableOfContents.vue

```diff
@@ -47,10 +47,10 @@ export default {
 	},
 	methods: {
 		goto(heading) {
-			document.getElementById(heading.id).scrollIntoView()
-
+			const element = document.getElementById(heading.id)
```

Comment:
```suggestion
			const contentContainer = this.$parent.$parent.$el
			const element = contentContainer.querySelector(`#${heading.id}`)
```

As mentioned, unfortunately `document.getElementById()` doesn't work if several editors with the ID are loaded. You can reproduce this by opening a `Readme.md` from files app while the same file is also loaded in the folder description editor.

My suggestion instead searches the `#${heading.id}` element from the grandparent of the Vue component, which makes sure that it's always taken from the editor instance we're operating on.

Reply:
```suggestion
			const element = this.$root.$el.querySelector(`#${heading.id}`)
```

Indeed, using `$root` is much better. Here's an updated suggestion that worked in my tests.